src/script/cpp_api/s_security.cpp

```diff
@@ -573,12 +573,28 @@ bool ScriptApiSecurity::checkPath(lua_State *L, const char *path,
 			if (mod) {
 				str = fs::AbsolutePath(mod->path);
 				if (!str.empty() && fs::PathStartsWith(abs_path, str)) {
+					if (write_required) {
+						const char *message =
+								"Writing to mod directories is deprecated, as any changes "
+								"will be overwritten when updating content. "
+								"Use minetest.get_mod_data_path() instead.";
+						log_deprecated(L, message, 1);
```

Comment:
FYI this warning is raised when generating `testnodes_generated_mb.png` and `testnodes_generated_ck.png`.

Reply:
Is generating textures at load time considered OK? I don't see a problem with it as long as they are generated anew every time.

Reply:
See todo in OP

Reply:
I suppose mods shouldn't do this as they might not have write access to the mod directory.